boot/seal.go

```diff
@@ -669,11 +561,38 @@ func recoveryBootChainsForSystems(systems []string, modesForSystems map[string][
 	return chains, nil
 }
 
-func runModeBootChains(rbl, bl bootloader.Bootloader, modeenv *Modeenv, cmdlines []string, runSnapsDir string) ([]BootChain, error) {
-	tbl, ok := rbl.(bootloader.TrustedAssetsBootloader)
-	if !ok {
-		return nil, fmt.Errorf("recovery bootloader doesn't support trusted assets")
+func runModeBootChains(rbl bootloader.TrustedAssetsBootloader, bl bootloader.Bootloader, modeenv *Modeenv, cmdlines []string, runSnapsDir string) ([]BootChain, error) {
+	if rbl == nil {
+		return runModeBootChainsUntrusted(modeenv, runSnapsDir)
```

Comment:
Previously we were returning errors when trusted assets were not supported, what does secboot for the untrusted case? Is this for when we use hooks and there is no TPM?